File: 1807-partitioning-into-minimum-number-of-deci-binary-numbers/partitioning-into-minimum-number-of-deci-binary-numbers.c
int minPartitions(char* n) {
    int ans = 0;
    for(int i=0; n[i] != '\0'; i++){
        int temp = n[i] - '0';
        ans = (ans>temp? ans:temp);
        if(ans==9)return ans;
    }
    return ans;
}

File: 1807-partitioning-into-minimum-number-of-deci-binary-numbers/partitioning-into-minimum-number-of-deci-binary-numbers.js
/**
 * @param {string} n
 * @return {number}
 */
var minPartitions = function(n) {
    let arr = n.split('');
    let maxi = arr.reduce((prev, curr)=>{
        return prev>curr?prev:curr;
    })
    return maxi - '0';
};